syncd/syncd_notifications.cpp

```diff
@@ -176,13 +173,34 @@ void redisPutFdbEntryToAsicView(
                 ]
             }]
             */
-            SWSS_LOG_ERROR("received a flush port fdb event, port_oid = 0x%" PRIx64 ", bv_id = 0x%" PRIx64 ", unsupported", port_oid, bv_id);
+            SWSS_LOG_NOTICE("received a flush fdb event, port_oid = 0x%" PRIx64 ", bv_id = 0x%" PRIx64, port_oid, bv_id);
+
+            auto strBridgePortId = sai_serialize_object_id(port_oid);
+
+            std::string pattern = ASIC_STATE_TABLE + std::string(":SAI_OBJECT_TYPE_FDB_ENTRY:*");
+            for (const auto &fdbkey: g_redisClient->keys(pattern))
+            {
+                /* we only remove dynamic fdb entries here, static fdb entries need to be deleted manually by user instead of flush */
+                auto pEntryType = g_redisClient->hget(fdbkey, "SAI_FDB_ENTRY_ATTR_TYPE");
+
+                if (pEntryType && *pEntryType == "SAI_FDB_ENTRY_TYPE_DYNAMIC")
+                {
+                    auto bp = g_redisClient->hget(fdbkey, "SAI_FDB_ENTRY_ATTR_BRIDGE_PORT_ID");
+
+                    if (bp && *bp == strBridgePortId)
+                    {
+                        SWSS_LOG_DEBUG("remove fdb entry %s for SAI_FDB_EVENT_FLUSHED",fdbkey.c_str());
+                        g_redisClient->del(fdbkey);
+                    }
+                }
+                else
+                {
+                    SWSS_LOG_ERROR("found unknown type fdb entry, key %s", fdbkey.c_str());
```

Comment:
With FLUSH_TYPE ALL,  should we also get flush notification for static?

```
    /** Flush static and dynamic FDB entries */
    SAI_FDB_FLUSH_ENTRY_TYPE_ALL,
```

Reply:
yes

Reply:
actually

Reply:
no

Reply:
this is notification processing, and notification don't get type type, that's why if vendor wants to flush all static and non static, then its required for vendor to send 2 flush notifications, one, when type is set to dynamic, and second for static

Reply:
since sai_fdb_event_notification_data_t contains sai_fdb_event_t event_type, and not sai_fdb_flush_entry_type_t

Reply:
fixed